fastavro/_reader.pyx

```diff
@@ -274,26 +280,26 @@ cpdef read_double(ReaderBase fo, writer_schema=None, reader_schema=None):
     """
     cdef bytes data
     cdef unsigned char ch_data[8]
-    cdef double_long dl
+    cdef double_ulong64 dl
     data = fo.read(8)
     if len(data) == 8:
         ch_data[:8] = data
         dl.n = (ch_data[0] |
-                (<unsigned long>(ch_data[1]) << 8) |
```

Comment:
I noticed this code was still using `unsigned long`. This may have been a bug on Windows, where `long` is 32 bits but `double` on Windows is 64 bits. I wonder why tests didn't catch it.

Reply:
It's possible that we don't have a test that uses the `double` type in such a way that would have exposed the problem.